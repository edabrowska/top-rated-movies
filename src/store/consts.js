export const GET_MOVIES = 'GET_MOVIES'
export const MOVIE_MODAL_OPEN = 'MOVIE_MODAL_OPEN'
export const MOVIE_MODAL_CLOSE = 'MOVIE_MODAL_CLOSE'
