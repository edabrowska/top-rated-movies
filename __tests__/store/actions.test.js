import configureStore from 'redux-mock-store'
import nock from 'nock'
import thunk from 'redux-thunk'

import initialState from '../../__mocks__/fileMock'

import {
	getMovies,
	openMovieModal,
	closeMovieModal
} from '../../src/store/actions'

import { API_KEY } from '../../src/utils/consts'

describe('action get data', () => {
	const middlewares = [ thunk ]
	const mockStore = configureStore(middlewares)
	const allMovies = initialState.app.movies
	const oneMovie = initialState.movieDetails.movie
	const movie_id = 419704
	let store

	beforeEach(() => {
		store = mockStore({})
	})

	afterEach(() => {
		// clear all HTTP mocks after each test
		nock.cleanAll()
	  })

	it('gets data', () => {
		nock('https://api.themoviedb.org/3')
			.get(`/movie/popular?api_key=${API_KEY}&page=1`)
			.reply(200, allMovies)

		return store.dispatch(getMovies()).then(() => {
			expect(store.getActions()).toMatchSnapshot()
		})
	})

	it('gets single data and opens modal', () => {
		nock('https://api.themoviedb.org/3')
			.get(`/movie/${movie_id}?api_key=${API_KEY}`)
			.reply(200, oneMovie)

		return store.dispatch(openMovieModal(movie_id)).then(() => {
			expect(store.getActions()).toMatchSnapshot()
		})
	})
})

describe('action close modal', () => {
	const mockStore = configureStore()
	const store = mockStore({})

	it('closes modal', () => {
		store.dispatch(closeMovieModal())

		expect(store.getActions()).toMatchSnapshot()
	})
})