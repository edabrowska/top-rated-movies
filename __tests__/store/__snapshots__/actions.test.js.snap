// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action close modal closes modal 1`] = `
Array [
  Object {
    "payload": null,
    "type": "MOVIE_MODAL_CLOSE",
  },
]
`;

exports[`action get data gets data 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "GET_MOVIES",
  },
]
`;

exports[`action get data gets single data and opens modal 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "id": 419704,
        "release_date": "2019-09-17",
        "title": "Ad Astra",
        "vote_average": 6,
      },
      Object {
        "id": 495764,
        "release_date": "2020-02-05",
        "title": "Birds of Prey (and the Fantabulous Emancipation of One Harley Quinn)",
        "vote_average": 7.1,
      },
    ],
    "type": "MOVIE_MODAL_OPEN",
  },
]
`;
